src/algorithms/machinelearning/tensorflowpredicteffnetdiscogs.cpp

```diff
@@ -0,0 +1,284 @@
+  _network = new scheduler::Network(_vectorInput);
+}
+
+
+void TensorflowPredictEffnetDiscogs::configure() {
+  _tensorflowPredictEffnetDiscogs->configure(INHERIT("graphFilename"),
+                                             INHERIT("savedModel"),
+                                             INHERIT("input"),
+                                             INHERIT("output"),
+                                             INHERIT("patchHopSize"),
+                                             INHERIT("lastPatchMode"),
+                                             INHERIT("batchSize"),
+                                             INHERIT("patchSize"));
+
+  _patchHopSize = parameter("patchHopSize").toInt();
+  _patchSize = parameter("patchSize").toInt();
+  _batchSize = parameter("batchSize").toInt();
+  _lastPatchMode = parameter("lastPatchMode").toString();
+  _lastBatchMode = parameter("lastBatchMode").toString();
+}
+
+
+void TensorflowPredictEffnetDiscogs::compute() {
+  const vector<Real>* signal = &_signal.get();
+  vector<vector<Real> >& predictions = _predictions.get();
+
+  if (!signal->size()) {
+    throw EssentiaException("TensorflowPredictEffnetDiscogs: empty input signal");
+  }
+
+  vector<Real> paddedSignal;
+  int paddingPatches;
+  if (_lastBatchMode == "zeros" || _lastBatchMode == "same") {
+    // Computes the number of patches required to fill the final batch and makes a
+    //  zero-padded copy of the input signal only when needed.
+    paddingPatches = padSignal(*signal, paddedSignal);
+    if (paddingPatches) signal = &paddedSignal;
+  }
+  _vectorInput->setVector(signal);
+
+  _network->run();
+
+  try {
+    predictions = _pool.value<vector<vector<Real> > >("predictions");
+    if (_lastBatchMode == "same") {
+      predictions.erase(predictions.end() - paddingPatches, predictions.end());
+    }
+  }
+  catch (EssentiaException&) {
+    predictions.clear();
+  }
+
+  reset();
+}
+
+
+void TensorflowPredictEffnetDiscogs::reset() {
+  _network->reset();
+  _pool.remove("predictions");
+}
+
+int TensorflowPredictEffnetDiscogs::padSignal(const std::vector<Real> &signal, std::vector<Real> &paddedSignal) {
+  int nSamples = signal.size();
+
+  // FrameCutter zero-pads the signal so that the first frame is zero-centered.
+  // Consider that to estimate the number of hops.
+  Real hops = ((Real)nSamples - (Real)_frameSize / 2.0) / (Real)_hopSize;
+
+  // By default FrameCutter, zero-pads the final samples to form a last frame.
+  int nFrames = 1 + ceil(hops);
+
+  // Frames to patches.
+  Real patchHops = ((Real)nFrames - (Real)_patchSize) / (Real)_patchHopSize;
+
+  int nPatches;
+  if (_lastPatchMode == "repeat") {
+    nPatches = 1 + ceil(patchHops);
+  } else if (_lastPatchMode == "discard") {
+    nPatches = 1 + floor(patchHops);
+  } else {
+    throw EssentiaException("TensorflowPredictEffnetDiscogs: incorrect `lastPatchMode`");
+  }
+
+  // Patches to batches.
+  int nBatches = ceil((Real)nPatches / (Real)_batchSize);
```

Comment:
`_batchSize` = 0 is accepted by the parameter definition `[-1,inf)`, so this would lead to division by zero.